### PR TITLE
Refactor cmd/sync to use new package frames

### PR DESCRIPTION
**Pull request by bartekpacia**
as the title says

@dsonyy please clone this locally and make sure it works fine (I don't have a working MPU6050)

### CODE REVIEW

File: cmd/sync/sync.go [hunk 1]
```diff
@@ -92,37 +92,32 @@ func main() {
...
 
-		// TODO: Clever Buffer -> Frame conversion
-		// TODO: Use frames.Assemble
-		frame := frames.Frame{
-			Header:   frames.FrameHeader(data.String()[0:3]),
-			Data:     []byte(data.String()[3:15]),
-			Checksum: byte(data.String()[16]),
-		}
+		// TODO: Make sure that it works correctly
+		frame := frames.Assemble(data[0:2], data[3:15], data[16])
```
Comment (bartekpacia) [inline cmd/sync/sync.go]: this is deliberate – header is `LD`, not `LD+`

**Comment by dsonyy** [comment]
```go
// Verify checks whether the frame is valid (i.e of correct format).
//
// The frame has to:
//
// - have exactly 1 plus sign ("+")
//
// - have exactly 1 hash sign ("#")
//
// - its checksum must be correct
func Verify(frame Frame) bool {
	if bytes.Count(frame, []byte{'+'}) != 1 {
		return false
	}

	if bytes.Count(frame, []byte{'#'}) != 1 {
		return false
	}

	checksum := CalculateChecksum(frame)
	return checksum == frame.Checksum()
}
```
@bartekpacia the code checks whether the entire frame contains only one `+` sign. What if a data part will have a byte with 43 value (`+` sign)? Then the frame will be considered as invalid. It should check if the byte just before the checksum byte is `+`.
(i know that it's not the right PR for that)

**Comment by dsonyy** [comment]
```go
// Header returns frame's header. It is usually 2 bytes.
func (f Frame) Header() []byte {
	end := strings.IndexByte(string(f), '+')
	return f[0:end]
}
```
@bartekpacia Panics, because for example accelerometer frames uses `-` sign, not `+`.

**Comment by bartekpacia** [comment]
1) You're right, I didn't think about it. Thanks – fixed.
2) I wanted to tell you but I must've forgotten – I'm suggesting using the `+` for frames flowing in both directions. This will make the code simpler, and tbh – do we really need this? I think not.

**Comment by bartekpacia** [comment]
aw shit.


`frames.Header()` and `frames.Data()`are also broken.

for example, this:

```go
// Data returns frame's data part from the first byte after a plus sign ("+") up
// to the antepenultimate (last but one - 1) byte.
func (f Frame) Data() []byte {
	start := strings.IndexByte(string(f), '+')
	return f[start+1 : len(f)-2]
}
```

is will not work correctly if data has plus sign (`+`)

first solution that came to my head:
change header from `LD` to `LDx`, where `x` is the number of data bytes. This way we'll exactly know where data part of a frame begins and ends.

**Pull request merged**